uk.co.compucorp.civicrm.hrleaveandabsences/js/angular/test/mocks/data/leave-request-data.js

```diff
@@ -1,101 +1,207 @@
+        'type_id': '2',
+        'contact_id': '202',
+        'status_id': '4',
+        'from_date': '2016-01-01',
+        'from_date_type': '1',
+        'to_date': '2016-01-01',
+        'to_date_type': '1'
+      }]
     },
     singleDataSuccess: {
-      "is_error": 0,
-      "version": 3,
-      "count": 1,
-      "values": [
-        {
-          "id": "17",
-          "type_id": "1",
-          "contact_id": "202",
-          "status_id": "1",
-          "from_date": "2016-02-01",
-          "from_date_type": "1",
-          "to_date": "2016-02-03",
-          "to_date_type": "1"
-        }
-      ]
+      'is_error': 0,
+      'version': 3,
+      'count': 1,
+      'values': [{
+        'id': '17',
+        'type_id': '1',
+        'contact_id': '202',
+        'status_id': '1',
+        'from_date': '2016-02-01',
+        'from_date_type': '1',
+        'to_date': '2016-02-03',
+        'to_date_type': '1'
+      }]
     },
     singleDataError: {
-      "is_error": 1,
-      "error_message": jasmine.any(String)
+      'is_error': 1,
+      'error_message': jasmine.any(String)
     },
     balanceChangeByAbsenceTypeData: {
+      'is_error': 0,
+      'version': 3,
+      'count': 3,
+      'values': {
+        '1': -21,
+        '2': -1,
+        '3': -11
+      }
+    },
+    calculateBalanceChangeRequestData: {
+      'contact_id': 109,
+      'from_date': '2016-11-05',
+      'from_date_type': optionGroupMock.randomValue('hrleaveandabsences_leave_request_day_type', 'name'),
+      'to_date': '2016-11-10',
+      'to_date_type': optionGroupMock.randomValue('hrleaveandabsences_leave_request_day_type', 'name')
+    },
+    calculateBalanceChangeData: {
+      'is_error': 0,
+      'version': 3,
+      'count': 2,
+      'values': {
+        'amount': -4,
+        'breakdown': [{
+          'date': '2016-11-05',
+          'amount': 0,
+          'type': {
+            'id': 4,
+            'value': 4,
+            'label': 'Weekend'
+          }
+        }, {
+          'date': '2016-11-06',
+          'amount': 0,
+          'type': {
+            'id': 4,
+            'value': 4,
+            'label': 'Weekend'
+          }
+        }, {
+          'date': '2016-11-07',
+          'amount': 1,
+          'type': {
+            'id': 1,
+            'value': 1,
+            'label': 'All Day'
+          }
+        }, {
+          'date': '2016-11-08',
+          'amount': 1,
+          'type': {
+            'id': 1,
+            'value': 1,
+            'label': 'All Day'
+          }
+        }, {
+          'date': '2016-11-09',
+          'amount': 1,
+          'type': {
+            'id': 1,
+            'value': 1,
+            'label': 'All Day'
+          }
+        }, {
+          'date': '2016-11-10',
+          'amount': 1,
+          'type': {
+            'id': 1,
+            'value': 1,
+            'label': 'All Day'
+          }
+        }]
+      }
+    },
+    randomLeaveRequestFromServer: {
```

Comment:
What's the point of this? You already have a collection of mocked leave requests, why don't you pick one at random?